src/nvim/drawscreen.c

```diff
@@ -1799,6 +1799,12 @@ static void win_update(win_T *wp, DecorProviders *providers)
       did_update = DID_NONE;
     }
 
+    if (wp->w_redr_numcol) {
+      wp->w_redr_numcol = false;
+      changed_line_abv_curs_win(wp);
+      goto win_update_start;
+    }
```

Comment:
We do a (recursive) redraw here to fix the following scenario(without this the first four lines would have a smaller number column until then next redraw):
[![asciicast](https://asciinema.org/a/0skefCZlEsjdKl88rCww0EJ3x.svg)](https://asciinema.org/a/0skefCZlEsjdKl88rCww0EJ3x)

Simply checking for `nrwidth < wp->w_nrwidth_width` causes some terminal/sign tests to fail for some reason. Seemed weird to me that tests related to terminal/signs would cause the numbercolumn width to change during a redraw, and that redrawing the window would cause the test to fail, or is it not? Anyways, I resorted to adding a boolean to the window structure instead.

I'm not sure if a `numbercolumn` format expression that makes sense(the example doesn't really of course) would ever cause this to happen but redrawing the window in the case that it does happen seemed more appropriate than truncating the number column to me.

Reply:
> extending the numbercolumn and redrawing the window seemed more appropriate than truncating the number column to me.

Shouldn't we take `'numberwidth'` into account? It allows up to 20 column. So the automatic value for minimum would be `'numberwidth'` (maybe minus 1, check `:help 'numberwidth'`) and the maximum would be 20. 🤔

Reply:
Yes, this is done in the current PR:
https://github.com/luukvbaal/neovim/blob/080dc0def7c47556ab4338607dffc21950347250/src/nvim/statusline.c#L552-L556